client/src/components/PermissionsEditor.vue

```diff
@@ -3,19 +3,22 @@
 		{{ $t("permissions-editor.title") }}<br />
 		{{ $t("permissions-editor.text1") }}<br />
 		{{ $t("permissions-editor.text2") }}<br />
-		{{ $t("permissions-editor.viewing-as") }}: {{ ROLE_DISPLAY_NAMES[currentRole] }}<br />
+		{{ $t("permissions-editor.viewing-as") }}: 
+		{{ $t(`roles.${currentRole}`) }}<br />
 		<v-table density="compact" :key="updateEpoch">
 			<thead>
 				<tr>
 					<th class="text-left" scope="col">{{ $t("permissions-editor.permission") }}</th>
-					<th class="text-left" scope="col" v-for="i in 5" :key="i">
-						{{ ROLE_NAMES[i - 1] ? ROLE_DISPLAY_NAMES[i - 1] : 0 }}
-					</th>
+					<th class="text-left" scope="col">{{ $t(`roles.${[Role.UnregisteredUser]}`) }}</th>
+					<th class="text-left" scope="col">{{ $t(`roles.${[Role.RegisteredUser]}`) }}</th>
```

Comment:
nit: Ok, I'm surprised this actually works as you have it. Weird. Either way, it should be without the square brackets

```vue
<th class="text-left" scope="col">{{ $t(`roles.${Role.RegisteredUser}`) }}</th>
```